xdsl/dialects/tosa.py

```diff
@@ -284,6 +284,57 @@ def verify_(self) -> None:
             )
 
 
+@irdl_op_definition
+class MaxPool2DOp(IRDLOperation):
+    """
+    TOSA dialect operation for performing 2D max pooling on a tensor.
+
+    See external [documentation](https://mlir.llvm.org/docs/Dialects/TOSA/#tosamax_pool2d-mlirtosamaxpool2dop).
+    """
+
+    name = "tosa.max_pool2d"
+
+    T: ClassVar = VarConstraint("T", AnyAttr())
+    input = operand_def(TensorType.constr(T))
+    output = result_def(TensorType.constr(T))
+
+    kernel = prop_def(DenseArrayBase[I64])
+    stride = prop_def(DenseArrayBase[I64])
+    pad = prop_def(DenseArrayBase[I64])
+    nan_mode = opt_prop_def(StringAttr)
+
+    def _verify_(self) -> None:
```

Comment:
Does this work? Can you also add the custom format to AvgPool2DOp? And rewrite the tests to use the custom format
```suggestion
    nan_mode = opt_prop_def(StringAttr)

    assembly_format = "operands attr-dict `:` functional-type(operands, results)

    def _verify_(self) -> None:
```